### ISSUE TITLE
If the method name is 'getValue'  and the return type is a '@FreeBuilder' interface type,the Autogenerated code has error.

### ISSUE THREAD

**Issue by rocsky** [google/FreeBuilder#78]
**environment**
eclipse Mars.1 Release (4.5.1)
freebuilder-1.2-rc1.jar
jdk1.8

**example**
```java
@FreeBuilder
public interface Value {
	class Builder extends Value_Builder{}
}

@FreeBuilder
public interface Person {
	
	Value getValue();
	 
	class Builder extends Person_Builder{
		public static Builder of(){
			return new Builder();
		}
		
		public static Builder of(Person metadata){
			return new Builder().mergeFrom(metadata);
		}
		
		public static Builder of(Builder builder){
			return new Builder().mergeFrom(builder);
		}
		
	}
}
```
**Person_Builder**  can't be compiled.

**Reason**
The reason is the variable **value** is ambiguous. as show below.

the Autogenerated method **mergeFrom**
```java

  /**
   * Sets all property values using the given {@code Person} as a template.
   */
  public Person.Builder mergeFrom(Person value) {
    value.mergeFrom(value.getValue());
    return (Person.Builder) this;
  }

```

**Comment by cjp39** [comment]
Ooooh, evil! Thanks :)

**Comment by hdurer** [comment]
The sad thing is that it's now almost 30 years that the concept of hygienic macros has been known (according to the Wikipedia page).

Maybe your next project?

**Comment by cjp39** [comment]
We're stymied a bit here because method parameter names are part of the JavaDoc, so we can't just obfuscate them.

**Comment by cjp39** [comment]
And don't forget, we also want readable generated code, so _all_ names must be as clear as possible.

**Comment by cjp39** [comment]
You can pick this fix up in v1.0.2. Thanks for the report!
